tools/clang/lib/Sema/SemaHLSL.cpp

```diff
@@ -15610,4 +15595,28 @@ void DiagnoseEntry(Sema &S, FunctionDecl *FD) {
   }
   }
 }
+
+// Check if intrinsic function is illegal in current shader model.
+bool isIllegalIntrinsic(const CXXMethodDecl *MD, IntrinsicOp opCode,
```

Comment:
nit: What about `isInvalidForHLSLTarget`?

"Illegal" sounds a bit strong wording and the name doesn't really capture that it is only contextually invalid.

Reply:
The function is removed.